src/ring_theory/ideal_operations.lean

```diff
@@ -474,6 +492,33 @@ le_antisymm (map_le_iff_le_comap.2 $ mul_le.2 $ λ r hri s hsj,
   by rw [← f.map_mul];
   exact mem_map_of_mem (mul_mem_mul hri hsj))
 
+@[simp] lemma map_id : I.map (ring_hom.id R) = I :=
+by ext; simp [ideal.map]
+
+lemma map_map {T : Type*} [comm_ring T] {I : ideal R} (f : R →+* S)
+  (g : S →+*T) : (I.map f).map g = I.map (g.comp f) :=
+le_antisymm begin
+  refine ideal.span_le.2 _,
+  rintros x ⟨y, hy, rfl⟩,
+  rw [submodule.mem_coe, ← ideal.mem_comap],
+  refine ideal.span_le.2 _ hy,
+  rintros y ⟨z, hz, rfl⟩,
+  rw [submodule.mem_coe, ideal.mem_comap, ← ring_hom.comp_apply],
+  exact ideal.mem_map_of_mem hz
+end
+(ideal.span_mono begin
+  rw [ring_hom.coe_comp, set.image_comp],
+  exact set.image_subset _ ideal.subset_span,
+end)
+
+variables {I f}
+lemma comap_map_le : I ≤ (I.map f).comap f :=
```

Comment:
Don't we have some `gc` lying around that will give you this in one line?
`by rw ← map_le_iff_le_comap; refl`?

Reply:
I've added this, and proved some of the existing lemmas using gc, sometimes by generalising them to gc first. I also transferred everything about gc to ideals.

Reply:
The current commit won't build, but the next one should.

Reply:
It should build now. I proved it was a galois insertion when `f` is surjective as well and transferred some lemmas there as well.